Ali | Amir / Storybook for components (#2009)
binary-com/deriv-app: packages/components/stories/auto-height-wrapper/README.md

# Auto Height Wrapper

> Renders a wrapper to control the height of elements. `AutoHeightWrapper` does not allow the child component's height to be greater than the client's screen size and makes sure that the children is always visible.

Reviewer: ```suggestion
A wrapper to control the height of its children. `AutoHeightWrapper` does not allow the child components' height to be greater than the client's screen size and makes sure that the children are always visible.
```

## Usage

```jsx
import AutoHeightWrapper from 'Components/auto-height-wrapper';

const DummyComponent = props => (
    <AutoHeightWrapper default_height={300} height_offset={50}>
        {({ setRef, height }) => (
            <div ref={setRef} style={{height}}/>
        )}
    </AutoHeightWrapper>
);
```

## Props

| Name            | Type            | Default          | Description                                             |
| --------------- | --------------- | ---------------- | ------------------------------------------------------- |

> | default_height  | {number}        | null             | Default height of the element.                          |

Reviewer: If it's a number why we set the default to `null`?

Author: There is no default value for `default_height`

| height_offset   | {number}        | 0                | Height offset for the element.                          |

> | setRef          | {function}      | null             | Function to choose the element to control it's height.  |

Reviewer: ```suggestion
| setRef          | {function}      | null             | Function to set a reference to the element which the wrapper controls its height.  |
```

| height          | {number}        | default_height   | Tunned height of the element.                           |

# Full example:

```jsx
import AutoHeightWrapper from 'Components/auto-height-wrapper';

const DummyComponent = props => (
    <AutoHeightWrapper default_height={300} height_offset={50}>
        ({ setRef, height }) => (
            <div ref={setRef} style={{height}}>       

>                 {Array.from(new Array(15)).map((item,index) => {
>                     return <p>This is the text number {index}</p>
>                 })}

Reviewer: ```suggestion
                {Array.from(new Array(15)).map((_,index) => <p>This is the text number {index}</p>)}
```
